Std/Tactic/Alias.lean

```diff
@@ -0,0 +1,204 @@
+/-
+Copyright (c) 2017 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro, David Renshaw
+-/
+import Lean
+
+/-!
+# The `alias` command
+
+This file defines an `alias` command, which can be used to create copies
+of a theorem or definition with different names.
+
+Syntax:
+
+```lean
+/-- doc string -/
+alias my_theorem ← alias1 alias2 ...
+```
+
+This produces defs or theorems of the form:
+
+```lean
+/-- doc string -/
+theorem alias1 : <type of my_theorem> := my_theorem
+
+/-- doc string -/
+theorem alias2 : <type of my_theorem> := my_theorem
+```
+
+Iff alias syntax:
+
+```lean
+alias A_iff_B ↔ B_of_A A_of_B
+alias A_iff_B ↔ ..
+```
+
+This gets an existing biconditional theorem `A_iff_B` and produces
+the one-way implications `B_of_A` and `A_of_B` (with no change in
+implicit arguments). A blank `_` can be used to avoid generating one direction.
+The `..` notation attempts to generate the 'of'-names automatically when the
+input theorem has the form `A_iff_B` or `A_iff_B_left` etc.
+-/
+
+namespace Tactic
+namespace Alias
+
+open Lean Elab Parser.Command
+
+/-- Adds some copies of a theorem or definition. -/
+syntax (name := alias) (docComment)? "alias " ident " ←" (ppSpace ident)* : command
+
+/-- Adds one-way implication declarations. -/
+syntax (name := aliasLR) (docComment)?
+  "alias " ident " ↔ " binderIdent ppSpace binderIdent : command
+
+/-- Adds one-way implication declarations, inferring names for them. -/
+syntax (name := aliasLRDots) (docComment)? "alias " ident " ↔ " ".." : command
+
+/-- Like `++`, except that if the right argument starts with `_root_` the namespace will be
+ignored.
+```
+appendNamespace `a.b `c.d = `a.b.c.d
+appendNamespace `a.b `_root_.c.d = `c.d
+```
+
+TODO: Move this declaration to a more central location.
+-/
+def appendNamespace (ns : Name) : Name → Name
```

Comment:
Maybe this should be moved to a central location in Std? Other than that, LGTM

Reply:
or in lean... I know this is reimplemented at least 3 times in lean core

Reply:
I'm happy to change this given a pointer to an implementation in core.

Reply:
I couldn't find this exact thing in core but I did simplify the definition a bit.

Reply:
There is now a PR in core: [lean4#2341](https://github.com/leanprover/lean4/pull/2341).